frontend/src/components/widgets/DataFrame/EditingState.test.ts

```diff
@@ -181,4 +181,48 @@ describe("EditingState class", () => {
       `{"edited_cells":{"0:0":"foo"},"added_rows":[{"0":"foo","1":"foo"}],"deleted_rows":[1]}`
     )
   })
+
+  it.each([
+    [
+      `{"edited_cells":{"0:0":"foo"},"added_rows":[{"0":"foo","1":"foo"}],"deleted_rows":[1]}`,
```

Comment:
Do we need some negative test cases here or i guess they can never happen? 
These are the test cases I am seeing:
Initial state:
2 rows.
1. edit first row/col, added a row, and deleted row 1
2. nothing
3. added a row
4. deleted row 1

From my perspective, I think we are missing a single edit test.
In addition, can the first test case be an edit on an added cell so that it's a little bit more complicated? 

> {"edited_cells":{"`2:0`":"bar"},"added_rows":[{"0":"foo","1":"foo"}],"deleted_rows":[1]}

Tho I think that might require a completely different test since i don't think you can just check the editing state because I think added rows would be this:
`added_rows":[{"0":bar","1":"foo"}]`

Reply:
Add an additional test for single edit. I don't think we need negative test cases here, since this structure is not controlled by the users.